### PR TITLE
[glsl-in] Refactors

### PR DESCRIPTION
**Pull request by teoxoy**
Refactors needed for https://github.com/gfx-rs/naga/pull/2309.

### CODE REVIEW

File: src/front/glsl/context.rs [hunk 1]
```diff
@@ -92,62 +94,80 @@ impl Context {
...
-                    .append(Expression::AccessIndex { base, index }, span);
+                let span = self.module.global_variables.get_span(handle);
+                let base = self.add_expression(Expression::GlobalVariable(handle), span)?;
+                let expr = self.add_expression(Expression::AccessIndex { base, index }, span)?;
 
                 (
                     expr,
                     {
-                        let ty = frontend.module.global_variables[handle].ty;
+                        let ty: Handle<Type> = self.module.global_variables[handle].ty;
```
[the remark this answers is not in the document]
Comment (teoxoy) [reply src/front/glsl/context.rs:163]: I think I misclicked to insert it (it was a type inlay).

**Review by jimblandy — APPROVED** [review]
Okay, looks good.

File: src/front/glsl/context.rs [hunk 2]
```diff
@@ -92,62 +94,80 @@ impl Context {
...
+    }
+
+    pub fn with_body<F>(&mut self, body: Block, cb: F) -> Result<Block>
+    where
+        F: FnOnce(&mut Self) -> Result<()>,
+    {
+        let old_body = std::mem::replace(&mut self.body, body);
+        let res = cb(self);
+        let body = std::mem::replace(&mut self.body, old_body);
+        res.map(|_| body)
```
[the remark this answers is not in the document]
Comment (jimblandy) [reply src/front/glsl/context.rs]: This can wait for a later change.

**Comment by jimblandy** [comment]
Pushed fixes for the comments I made earlier.

**Comment by jimblandy** [comment]
Slight reordering of commits, because I'd like to not squash this.

**Comment by jimblandy** [comment]
Rebased on 5329aa2b

**Pull request merged**